### DISCUSSION TITLE
IB TWS Fundamentals API Support

### DISCUSSION THREAD

**Discussion by damien93** [quantbelt/ib_fundamental#11] [General]
Hi there,

Supposedly IBs fundamental data has not been supported for years, according to their support chat. However I've been using it for years fine. 

My coding ability is nowhere near your level FYI. I use reqFundamentalData with reportType 'ReportsFinStatements', where i then parse the xml data and store everything i need in a dataframe. I merge it with price data and create screens based on fundamental metrics.

This past week I've found that the reportType 'ReportsFinStatements' is not returning anything for any stock/contract. Although other report types (e.g. ReportSnapshot) still seem to return data. Have you had this issue as well?

As an aside, are you concerned about the lack of support for the fundamentals data in IBs API?

Thanks for any response you can give, I appreciate your time.

@gnzsnz

**Comment by gnzsnz** [comment]
I have noticed intermittent failures, but so far (and i knock on wood) it has always returned.

I built `ib_fundamental` when the API was still supported, and did not published it because it stopped been supported. However as it kept working i decided to polish `ib_fundamental` and publish it. I'm aware that it will stop working one day. while it works i will use it...

**Comment by damien93** [comment]
Yeah I'm in the same boat. I'll use it for as long as it's around. I can't remember the last time I had a stretch of a few days where the fundamental requests (report type ReportsFinStatements specifically) weren't working, hopefully they come back after the weekend.

Do you have any other fundamentals data sources lined up as a back-up in case the IB data fails permanently some day? The only global one that looks to have a lot of depth and isn't too expensive that I know of is eod hd (https://eodhd.com/). I subscribed to it temporarily a while back to check it out but never built anything with it, so i'm not sure about the reliability or accuracy of it. Although I use their pricing data which has been good.

**Comment by miketraynor** [comment]
Did anyone work out what the issue was in the end? I have this issue now due to the forced upgrade from 10.19 in March 20th 2025. Oddly its just the ReportsFinStatements report that does work. The others are still ok. In addition to python, I tried various combinations of APIs in C# too but the ReportsFinStatements doesn't work there either. All versions of APIs fail on this report (10.30, 10.34 and older APIs) so it appears this might be due to the TWS upgrade itself :(

Does this currently work for anyone via Python (or any other language) at the moment?

Thanks

**Reply by damien93** [reply]
I'm only using python but seems it doesn't work anymore for ReportsFinStatements. I was using gateway 10.23 because it still worked on that, but not on newer versions. Now 10.23 no longer works at all. I've been using the fundamentals from ReportsFinStatements for years. If anyone knows if there's any way in which it can still work, would be greatly appreciated. Otherwise I've turned to an alternate data provider but the accuracy and timeliness of the data for markets other than the US is not nearly as good.
